rockusb/src/libusb.rs

```diff
@@ -462,3 +474,19 @@ where
         Ok(self.offset)
     }
 }
+
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub enum BootMode {
+    Maskrom = 0,
+    Loader = 1,
+}
+
+impl From<u8> for BootMode {
+    fn from(value: u8) -> Self {
+        match value {
+            0 => BootMode::Maskrom,
+            1 => BootMode::Loader,
+            _ => panic!("cannot convert {} to BootMode", value),
+        }
```

Comment:
A generic `From<u8> ` doesn't seem to be good idea (it's not any byte you can convert) also From really shouldn't fail

Reply:
ah yes, I intended to use `TryFrom` here. I will go by your recommendations, and implement a  'if-else' statement.